app/src/main/java/mozilla/lockbox/store/TelemetryStore.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.store
+
+import android.content.Context
+import android.preference.PreferenceManager
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import mozilla.lockbox.BuildConfig
+import mozilla.lockbox.R
+import mozilla.lockbox.action.TelemetryAction
+import mozilla.lockbox.extensions.filterByType
+import mozilla.lockbox.flux.Dispatcher
+import org.mozilla.telemetry.Telemetry
+import org.mozilla.telemetry.config.TelemetryConfiguration
+import org.mozilla.telemetry.event.TelemetryEvent
+import org.mozilla.telemetry.net.HttpURLConnectionTelemetryClient
+import org.mozilla.telemetry.ping.TelemetryCorePingBuilder
+import org.mozilla.telemetry.ping.TelemetryMobileEventPingBuilder
+import org.mozilla.telemetry.schedule.jobscheduler.JobSchedulerTelemetryScheduler
+import org.mozilla.telemetry.serialize.JSONPingSerializer
+import org.mozilla.telemetry.storage.FileTelemetryStorage
+
+open class TelemetryWrapper() {
```

Comment:
This should be an `interface` with an implementation both in actual code and in the tests for DI.

Reply:
This feels an awful lot like "intefaces for interfaces sake", which is a Java anti-pattern.

I can make that change, but I'd like a really good reason why it's necessary over having this default implementation with the testing's `FakeTelemetryWrapper` overriding stuff.